sfa_dash/blueprints/dash.py

```diff
@@ -21,6 +22,75 @@ def template_args(self, **kwargs):
     def get_site_metadata(self, site_id):
         return handle_response(sites.get_metadata(site_id))
 
+    def set_site_or_aggregate_metadata(self):
+        """Searches for a site_id or aggregate_id  in self.metadata
+        and loads the expected metadata object from the api in either
+        the 'site' or 'aggregate' key. If the object could not be retrieved,
+        sets a warning, sets the 'plot' template argument to None and
+        reraises the DataRequestError.
+        """
+        if self.metadata['site_id'] is not None:
+            try:
+                self.metadata['site'] = self.get_site_metadata(
+                    self.metadata['site_id'])
+            except DataRequestException:
+                self.temp_args.update({
+                    'warnings': {
+                        'Site Access': [
+                            'Site inaccessible. Plots will not be displayed.']
+                    },
+                })
+                raise
+        else:
+            try:
+                self.metadata['aggregate'] = handle_response(
+                    aggregates.get_metadata(self.metadata['aggregate_id']))
+            except DataRequestException:
+                self.temp_args.update({
+                    'warnings': {
+                        'Aggregate Access': [
+                            'Aggregate inaccessible. Plots will not be '
+                            'displayed.']
+                    },
+                })
+                raise
+
+    def set_site_or_aggregate_link(self):
+        """Creates a url to the forecast's site or aggregate. For injection
+        into the metadata template.
+
+        Returns
+        -------
+        link_html: str
+            An anchor tag referencing the site/aggregates page on the
+            dashboard or a string uuid if the metadata wasn't not available,
+            indicating that the current user does not have access to the
+            site or aggregate.
+        """
+        if self.metadata['site_id'] is not None:
```

Comment:
if you use ``self.metadata.get('site_id')`` here and below for aggregate_id, this should work just as well for observations